### ISSUE TITLE
Custom Inventory Upgrades Not Working

### ISSUE THREAD

**Issue by Delta-86** [benjee10/HabTech2#21]
I'm working on a compatibility patch for USI-LS and I wanted to try making recyclers installed via the utility rack system, but making the parts and additions to InventoryUpgrades.cfg leads to nothing happening.

Is the problem that multiple modules cannot be added by a single upgrade or that USI-LS itself can't function in this way?

Also, to make sure the patch only runs when USI-LS is installed, should NEEDS be added to the Upgrades node or each individual Module?

[InventoryUpgrades.txt](https://github.com/user-attachments/files/16679239/InventoryUpgrades.txt)

**Comment by benjee10** [comment]
Yes, as it stands only one module can be added per upgrade, however Vali is working on an overhaul of the system that will allow much more complex behaviour. I would hold off on any custom configs for now as the config structure is likely to change significantly; treat what's currently in the mod as a prototype.

I am also planning to add support for life support converters through TAC so you may be able to repurpose the parts/configs from that once it's complete.

I do notice that some of the USI modules rely on a moduleIndex which may cause problems even with multiple modules being added per upgrade, as the moduleIndex will change depending on what upgrades are installed. The stock science lab has this same issue.
